framework/doc/content/source/problems/FEProblemBase.md

```diff
@@ -147,10 +149,16 @@ $\psi_8$
 - MOOSE `Kernels` must provide each of the terms in square brackets (evaluated at $\vec{x}_{q}$ or $\vec x_{q_{face}}$ as necessary).
 
 
-!syntax description /Problem/FEProblem
-
 !syntax parameters /Problem/FEProblem
 
+!alert note
+When [!param](/Problem/FEProblem/check_uo_aux_state) is set to true, MOOSE will evaluate user objects
+(including all postprocessors and vector postprocessors) and auxiliary kernels on every execute flag except `linear`
```

Comment:
Why is `linear` treated specially? Is it because of the PJFNK check we have in libMesh? It seems like linear would be almost the most concerning if there are issues there

Reply:
This just because `FEProblemBase::computeResidualTags` is not calling `FEProblemBase::execute(EXEC_LINEAR)` function. Same as `FEProblemBase::computeResidualAndJacobian`. I think for residual (or linear), we already had a check in place, similar like this.

Reply:
Typically I think residual dependency issue in residual evaluations is caused by uo/auxkernels but not by kernels (unless kernels coupling tagged residual vectors, which we should prevent), so the existing check can be removed if we can call `execute(EXEC_LINEAR)`. This can be done along with the other `setup` cleanup in the future.

Reply:
Sad that `computeResidualTags` doesn't call `execute`